mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala

```diff
@@ -50,4 +59,75 @@ class CheckCmd(name: String = "check", description: String = "Check a TLA+ speci
   var view: String =
     opt[String](name = "view", description = "the state view to use with --max-error=n, default: transition index",
         default = "")
+
+  def setTuningOptions(): Map[String, String] = {
+    val tuning =
+      if (tuningOptionsFile != "") loadProperties(tuningOptionsFile) else Map[String, String]()
+    overrideProperties(tuning, tuningOptions)
+  }
+
+  def run() = {
+
+    val tuning = setTuningOptions()
+
+    logger.info("Tuning: " + tuning.toList.map { case (k, v) => s"$k=$v" }.mkString(":"))
+
+    executor.passOptions.set("general.tuning", tuning)
+    executor.passOptions.set("checker.nworkers", nworkers)
+    executor.passOptions.set("checker.discardDisabled", discardDisabled)
+    executor.passOptions.set("checker.noDeadlocks", noDeadlocks)
+    executor.passOptions.set("checker.algo", algo)
+    executor.passOptions.set("checker.smt-encoding", smtEncoding)
+    executor.passOptions.set("checker.maxError", maxError)
+    if (view != "")
+      executor.passOptions.set("checker.view", view)
+    // for now, enable polymorphic types. We probably want to disable this option for the type checker
+    executor.passOptions.set("typechecker.inferPoly", true)
+    setCommonOptions()
+    executor.run() match {
+      case Right(_)   => Right("Checker reports no error up to computation length " + length)
+      case Left(code) => Left(code, "Checker has found an error")
+    }
+  }
+
+  private def loadProperties(filename: String): Map[String, String] = {
```

Comment:
Note how this refactor lets us move these private methods that are only relevant for the `CheckCmd` into a module that only has to do with the `CheckCmd`.